### DISCUSSION TITLE
Fetch (method:Get) , issues with hash password

### DISCUSSION THREAD

**Discussion by Wolff51** [directus/directus#12972] [Archive]
Hello there,

I'm new in the dev game, and i'm trying to create my new first community website for rpg lovers.

I just create a register form and that's working well. 

I'm trying to create my form login, and check if the user exist. My fetch look like this : 

myAPIadress + /items/users?filter[email][_eq]=${email}&filter[password][_eq]=${password}

The thing is, i send a not hash password so it's not working. But i can't find in the doc how can i manage to make this work.
Request is working, it's just sending me an data object empty cause he can't find anything if the password is not hash.

Is there any method i didn't find in the doc to make this work ?

Thanks a lot for you help !

Edit : I juste use a simple text input, then i hash myself with bcrypt.

**Comment by rijkvanzanten** [comment]
You wouldn't retrieve the user based on the email and password combination, as you don't know the stored hash (on purpose). You'd fetch the user by a unique property (like the email), and then compare the hashed password locally, so you get something like (pseudo code):

```js
const user = await request('/users?filter[email][_eq]=rijkvanzanten@me.com&fields=email,password_hash');

bcrypt.verify(payload.password, user.password_hash).
```
